### ISSUE TITLE
[import module /sql] import additional directory causes dt to stall

### ISSUE THREAD

**Issue by AxelG-DE** [darktable-org/darktable#15861]
### Describe the bug

First of all: I have a little cold, so if there is inconsistency (or toooo long explanation) in my report, maybe related, as my concentration capability is low... BIG SORRY

When (trying to) importing an additional directory (with my Play Raw pics), dt including the gui stalls severely, so that I had to sigkill dt after 10 minutes or more (and I did restore the *.db from backup after each sigkill)

What I mean with gui stalls, please see screenshot below. "stall" means one one cpu-core loads to ~100% and no other things can be done on dt (other than creating nice screen-art by frozen gui, see below)

One default bug-report-analysis is to check with new config-dir. To not lose time prior 4.6 (if my issue is a bug at all) I report here prior such test (as I have >90k pictures to import). The result of new cofig-dir will be ammended as soon as possible (I will trigger that import after the initial bug report but then go to bed)

**Background / additioinal content**
@jenshannoschwalm asked me on IRC to report my file-system or directory situation:
- my pics are at /home/gerber/DATA/Bilder 2.4TB in size (was a separate disk in the past mounted to ~/DATA)
- for historical reasons I import the soft link: /home/gerber/Bilder
- this DATA folder is on an lvm with lvm-cache (old SSD) /dev/mapper/vg01-home--DATA on /home/gerber/DATA type ext4 (rw,nosuid,nodev,relatime,stripe=56)

- My Play RAWs are here: /home/Datengrab/Datengrab_Bilder/PlayRAW with 2.1G
- Other RAWs from friends are on level up in parallel to Play Raw at   /home/Datengrab/Datengrab_Bilder with 13G in total
- this folder is on the same physical HDD on lvm with lvm-cache on /dev/mapper/vg01-Datengrab on /home/Datengrab type ext4 (rw,nosuid,nodev,relatime,stripe=32)
- ~/ with ~/.config and ~/.cache is on an nvme with xfs

Permissions are fine!

**Issue**
just clicking on import/add to library makes dt and gui stall for 18s and even longer if "ignore-non-raw images" is not ticked. Ticking that takes a fortune again.

-d all -d perf delivers many, many lines like this: `70.8641 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1`
Sometimes even that stalls for minutes and suddenly creates another gazillions of such lines

in "places" of the "add to library" I do have old and partially orphaned directories. Clicking on those leads to a total hang for ever.

Just now I was patient and "Datengrab_Bilder" got added to places, but what I see in the proposed imports (the file list) makes no any sense to me (I discover this right now while typing this issue)
![grafik](https://github.com/darktable-org/darktable/assets/49004103/cbc52d05-02c7-4f52-91b4-5478d1faa50f)
that is all garbage IMHO.
1) I will not import mails
2) DATA/Bilder is in my homedir but I want to import "Datengrab_Bilder"

At this point I cancel. 
The output of -d all -d perf is below

### Steps to reproduce

As described above
1) Go to lighttable
2) click on import / add to library
3) change to another (accessible), in my case different mount point

See error

### Expected behavior

be as swift as it was before (open the import dialog lasted ~5s or so IIRC)

### Logfile | Screenshot | Screencast

Output of -d all -d perf
[dt-stall.txt.zip](https://github.com/darktable-org/darktable/files/13652731/dt-stall.txt.zip)

Gui frozen/stalled
![grafik](https://github.com/darktable-org/darktable/assets/49004103/f3939ef3-86be-463d-96e5-d0b0d3ac3439)

### Where did you obtain darktable from?

self compiled

### darktable version

darktable 4.5.0+1428~g81f018af64

### What OS are you using?

Linux

### What is the version of your OS?

gentoo

### Describe your system?

* darktable version : 4.5.0-1428-g81f018af64
* OS                : Linux - kernel 6.6.6-gentoo
* Distro            : Gentoo Base System release 2.14
* Processor         : Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
* Memory            : 32 GB (4 x 8 GB) + 5GB Swap
* Graphics card0    : NVIDIA GeForce GTX 1060 6GB
* Graphics card1    : NVIDIA GeForce RTX 2070 SUPER
* Graphics driver   : nvidia-drivers-535.129.03
* OpenCL installed  : Yes (opencl-headers-2023.04.17)
* OpenCL activated  : Yes
* Xorg              : xorg-server-21.1.9
* Desktop           : KDE 5
* GTK+              : gtk+-3.24.38
* gcc               : 13.2.1_p20230826
* cflags            : CMAKE_FLAGS="-march=native-O2-mtune=native-pipe"
* CMAKE_BUILD_TYPE  : "Release"

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

pls see above

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

please see above

**Comment by AxelG-DE** [comment]
I am just starting to import into new config-dir and behaviour is likely same
frozen gui
![grafik](https://github.com/darktable-org/darktable/assets/49004103/423b7f21-fe01-4728-b826-9363edf67485)

**Comment by AxelG-DE** [comment]
```
20.5748 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5749 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5750 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5751 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5751 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5752 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5756 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5757 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5840 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5841 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5842 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5842 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5844 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5844 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5847 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5849 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5852 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5852 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5855 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5857 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5859 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5860 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5861 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5862 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5863 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5864 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5864 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5865 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5866 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5868 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5868 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5869 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5871 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5872 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5874 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5884 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5885 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5885 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5886 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"
    20.5887 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"

```

**Comment by AxelG-DE** [comment]
I think I can stop here already and bring my original .config and .cache in place again

**Comment by gi-man** [comment]
I see a module called `selected images path` . That's not part of dt. Is that a LUA module?

**Comment by wpferguson** [comment]
Yes, it's from official/image_path_in_ui.lua

**Comment by jenshannoschwalm** [comment]
@TurboGit and @victoryforce i looked at this for short and from my understanding it might be related to checking to include all non-raws instead of jpegs. Unfortunately I am very short of time, could you check?

1. While checking for another group header we now would now have to exclude all non-raws.
2. While traversing all files we might want to exclude hidden files or backups (starting with '.'  or "~")?

**Comment by TurboGit** [comment]
@victoryforce : If you have some spare cycle, please have a look. Personally I won't have time before the full freeze in about 30h.

Looking at the report I'm wondering if there is not a recursive directory structure with some symlinks. I was not able to reproduce on my side anyway.

**Comment by TurboGit** [comment]
@AxelG-DE : BTW, is that a regression compared to 4.4.2? If you could check that would be very helpful. TIA.

**Comment by jenshannoschwalm** [comment]
I also couldn't reproduce so asking the OP for dirs. @AxelG-DE could you provide a tree for the "hanging" dirs?

**Comment by zisoft** [comment]
My guess here is a circular symbolic link from a directory to a directory somewhere higher in the same tree.

**Comment by AxelG-DE** [comment]
> @AxelG-DE : BTW, is that a regression compared to 4.4.2? If you could check that would be very helpful. TIA.

@TurboGit Just a few weeks ago I have re-imported all my 92k photos and all worked like a charm. In July and October I was on trips to Paris and Berlin and both set I imported with no issues. Hence I said on IRC/Matrix, yes, that I do see a regression and I will start now with revert or bisect around 8c1f7bbc9882f5fac80f509de8d10e985bd5ba88 and d67d557f0a7c9c928ae1faf78d7a6b7285a31390.



> @AxelG-DE could you provide a tree for the "hanging" dirs?

@jenshannoschwalm yesterday I put ~/.config/darktable and ~/.cache/darktable out of the way and wanted to import my standard picture dir and dt stalled for many minutes (I sigkilled afterwards). 
When you say "provide a tree" you mean 'find ./' ?
Technically I could, however, I would disclose a lot due to my naming scheme. Doesn't feel nice. I could give it to you personally instead, hence my approach on IRC-PM. However you said you have limited time, so I don't wanna put additional load to you.

I did check on my own. I typically do not use special characters, but I do have an Archive of my Phone-pics in the same dir and there I found pics with Chinese characters and at least one file with this name:` ./Archiv/Samsung_S3/Sicherheitskopie/Recovered_Pictures_2014-04-29/[000139].jpg` that is obviously odd with the [ ] but both of those files they are there since years.




> My guess here is a circular symbolic link from a directory to a directory somewhere higher in the same tree.

@zisoft no, I do not have such weird link structure and as I said, my picture dear is years old. 
![grafik](https://github.com/darktable-org/darktable/assets/49004103/825b7da9-542f-421b-a822-a0630a3b9d8c)

if you worry about two times "Bilder"... nope, that one is a Desktop-Icon
![grafik](https://github.com/darktable-org/darktable/assets/49004103/b05a7c7f-6cd6-4604-a308-61e0f1aec914)

**Comment by AxelG-DE** [comment]
4.4.2 is rather o.k. with importing, but even this is darktable 4.5.0+789~g07e1b26d98-dirty shows the effect 
(donno why is called dirty, I never fiddled with the code)

I will have a fresh clone and try again

**Comment by victoryforce** [comment]
> it might be related to checking to include all non-raws instead of jpegs.

> @victoryforce : If you have some spare cycle, please have a look.

@TurboGit Regarding the PR that replaced ignoring jpegs with ignoring non-raws, I looked carefully at the changes made and can't imagine how this could cause such a regression.

@AxelG-DE Ah, I noticed your comment that the regression was already present in a much earlier git snapshot (4.5.0+789), so it's definitely not from replacing the ignore criteria on import.

**Comment by TurboGit** [comment]
@victoryforce : Indeed, my guess was wrong. @AxelG-DE is doing a bisect as we speak, so we'll know the culprit soon hopefully.

**Comment by AxelG-DE** [comment]
hopefully....
bisect is hard, as db-updates forces me to del the dbs and also everytime dt stalls, I struggle, how long to wait

I do face the scenario, that one cpu-core is fully loaded, no CLI output at all, no HDD-io, GUI frozen...

**Comment by TurboGit** [comment]
> can it be, my sqlite is broken???

Could be, at this point no dev have been able to reproduce and the issue is not understood...

**Comment by AxelG-DE** [comment]
here we go (and especially the last two steps felt quite consistent as the crashes got more severe or stalls happened even earlier in the testing)

```
0122fd92b09d1a21fbaa0fd132cec14e2a1633e8 is the first bad commit
commit 0122fd92b09d1a21fbaa0fd132cec14e2a1633e8
Author: Pascal Obry <pascal@obry.net>
Date:   Sat Sep 30 22:15:51 2023 +0200

    Ensure ui_last/import_last_directory is sanitized.
    
    Fix a possible crash as reported in #15334.

 data/darktableconfig.xml.in         |  4 ++--
 src/control/conf.c                  | 22 ++++++++++++++++++++++
 src/control/conf.h                  |  2 ++
 src/libs/import.c                   | 12 ++++++++----
 tools/generate_darktablerc_conf.xsl | 16 +++++++++-------
 5 files changed, 43 insertions(+), 13 deletions(-)
```

**Comment by AxelG-DE** [comment]
I started with sqlite-3.43.2
finally also tested 3.42.0 and 3.44.2

all with the same result

**Comment by AxelG-DE** [comment]
just for the record. When ever I stall dt by opening the import dialog and try to change a folder to a different one, the Gui freezes and also the CLI output stops usually at approx 21s and only those sql-lines show up, even I started with -d all. Typically I wait for several minutes before I sigkill....
![grafik](https://github.com/darktable-org/darktable/assets/49004103/b732b74f-b1f2-456b-858a-130967e4b5b9)

**Comment by AxelG-DE** [comment]
```
$ git bisect log
git bisect start
# Status: warte auf guten und schlechten Commit
# good: [e44f1001a805324ca60fb7dba99764de53b04902] RELEASE_NOTES.md: Initialize for 4.6 release.
git bisect good e44f1001a805324ca60fb7dba99764de53b04902
# Status: warte auf schlechten Commit, 1 guter Commit bekannt
# bad: [88819c341e0328ab570ea2ab1c2e3ebb1ef0519d] Update ja.po (#15866)
git bisect bad 88819c341e0328ab570ea2ab1c2e3ebb1ef0519d
# good: [c2ca752f1f647ea40ad24e31b6df2ed2bd953b60] control & jobs maintenance
git bisect good c2ca752f1f647ea40ad24e31b6df2ed2bd953b60
# bad: [dfd848417b155ad9e41d3565e9816e6a49b6533c] fix snapshots with different crop
git bisect bad dfd848417b155ad9e41d3565e9816e6a49b6533c
# bad: [8f7465fa5dd7000d19199b1a85d932ddbd05ecad] Extended GPS coordinate import to support just decimal value in .xmp file (#15443)
git bisect bad 8f7465fa5dd7000d19199b1a85d932ddbd05ecad
# bad: [0397d6a1164b3282a72aed3d798b2c78d4b6fc94] Merge pull request #15401 from victoryforce/rating-range-filter-optimize-tooltip
git bisect bad 0397d6a1164b3282a72aed3d798b2c78d4b6fc94
# bad: [f67e6db257028b59b3112f52803a22a6e2056149] image: Minor code simplification for _any_altered_data.
git bisect bad f67e6db257028b59b3112f52803a22a6e2056149
# bad: [dbf6b68ef588e98b17c6cc127baa65c4db2092f4] Merge pull request #15364 from dterrahe/refactor_zoom
git bisect bad dbf6b68ef588e98b17c6cc127baa65c4db2092f4
# bad: [646b9e3f80cb33a8549ff3538d541f9a1e891e08] reduce cairo scale in post_expose
git bisect bad 646b9e3f80cb33a8549ff3538d541f9a1e891e08
# good: [4fc03872b0d11de76751ef535f325ba1f72801b7] RELEASE_NOTES.md: Add some entries for 4.6.
git bisect good 4fc03872b0d11de76751ef535f325ba1f72801b7
# good: [af0975cca0433995fc35d26eef14746d78095a4b] Unbreak build env preparation by removing the non-installable openexr package
git bisect good af0975cca0433995fc35d26eef14746d78095a4b
# bad: [07e1b26d987101daa2439daffaa53d17f22ddd80] Updates to pt_BR
git bisect bad 07e1b26d987101daa2439daffaa53d17f22ddd80
# bad: [0122fd92b09d1a21fbaa0fd132cec14e2a1633e8] Ensure ui_last/import_last_directory is sanitized.
git bisect bad 0122fd92b09d1a21fbaa0fd132cec14e2a1633e8
# first bad commit: [0122fd92b09d1a21fbaa0fd132cec14e2a1633e8] Ensure ui_last/import_last_directory is sanitized.
```

**Comment by TurboGit** [comment]
Hum... one of my commit... that's not possible I never introduce bug :)

What's your current value for `ui_last/import_last_directory` ?

**Comment by TurboGit** [comment]
@AxelG-DE : Thanks for the hard work, this certainly will help fixing the issue.

Can you check if #15870 works on your side? TIA.

**Comment by AxelG-DE** [comment]
> Hum... one of my commit... that's not possible I never introduce bug :)
> 
> What's your current value for `ui_last/import_last_directory` ?

we are not 100% sure yet, it is really a bug :-)

During my bisect I actually never completed an import. Remember darktablerc was fresh as well

my own darktablerc says `ui_last/import_last_directory=/home/gerber` where I expected /home/gerber/Bilder =8-/ 
This part I cannot explain, as I never import my home dir, I am not that boon. Again, during testing I used fresh configs (which are gone now)

Sorry to say, unfortunately #15870 does not help with my genuine .config and .cache and also not with fresh configs :-(
Today KDE even gave me a popup "darktable does not react kill/wait". But all over sudden dt kind of recoverd and the import dialog is closed. CLI gives me:
![grafik](https://github.com/darktable-org/darktable/assets/49004103/145794ce-1b3e-411a-9fe1-654c31103801)


BTW: also those commits I have selected as "good" yesterday, they also had that stall, where the CPU is blocking one core and have the line ```film_rolls WHERE folder = ?1"
    23,3968 [sql] /home/gerber/.local/git/darktable/src/common/film.c:102, function dt_film_get_id(): prepare "SELECT id FROM main.film_rolls WHERE folder = ?1"``` with no further output on the screen, but it recovers in a few seconds

IMHO there is something bigger going on... I desperately want output when CPU core is 100% and nothing else happens...

I will eject the code of #15870 again, so I can follow master

**Comment by AxelG-DE** [comment]
I had delted my former comment as I found an error during my this morning tests, **sorry!!!**

**Comment by AxelG-DE** [comment]
I will repeat the test!

**Comment by AxelG-DE** [comment]
> Hum... one of my commit... that's not possible I never introduce bug :)
> 
> What's your current value for `ui_last/import_last_directory` ?

We are not sure yet, it is a bug :-) :-)

import_last_directory was "/home/gerber" which is odd "/home/gerber/Bilder". I am not that stupid to import my entire home-dir.

Meanwhile we are online talking on IRC. 

For docu-purpose:
commit a9615b3292 almost no improvement
commit b4cce7b795 is way better and recovers much faster (still I can make dt stall for <300s)

For the first attempt in selecting a folder it already takes ~30s idle time, where is 100% load on one core and no IO
![grafik](https://github.com/darktable-org/darktable/assets/49004103/daff989c-6532-4b18-8655-1ff740fc71a7)

**Comment by jenshannoschwalm** [comment]
I think we fallback to home if the last config dir is not a dir!

**Comment by jenshannoschwalm** [comment]
We should avoid traversing hidden directories I think.

**Comment by TurboGit** [comment]
> We should avoid traversing hidden directories I think.

That's already the case.

**Comment by jenshannoschwalm** [comment]
> That's already the case.

I thought so too but in the log screenshot we see files tested in `/home/gerber/.local...` and `.local` should be avoided as hidden.

**Comment by TurboGit** [comment]
Strange as we have in the code:

https://github.com/darktable-org/darktable/blob/3c31b773aee7173380bdc5632629fec66cf1a504/src/libs/import.c#L688

**Comment by AxelG-DE** [comment]
Is the import dialog really searching pictures in .local? Maybe not and maybe it is just the path to the source, which in my case actually resides in .local
[dir_of_local_b.txt](https://github.com/darktable-org/darktable/files/13684246/dir_of_local_b.txt)

**Comment by AxelG-DE** [comment]
Some more explanations: 
I happily use [dt.bldr](https://github.com/jade-nl/dt.bldr)

~/.local/git holds my git repo of dt
~/.local/darktable/bin holds my self compiled master of dt
(the things in .local/share, I forgot, where they came from :-) )

**Comment by AxelG-DE** [comment]
today with version darktable 4.7.0+3~g9491da2c1c

I just clicked on "Datengrab_Bilder" in "places" and I got:
![grafik](https://github.com/darktable-org/darktable/assets/49004103/dce09003-ebc7-4f92-8bcf-ea4a5e317cdc)

**Comment by AxelG-DE** [comment]
Good thing is, it reconciles within ~20s

**Comment by TurboGit** [comment]
> Good thing is, it reconciles within ~20s

Ok, so that's because there is a huge list of directories/files to walk through.

**Comment by AxelG-DE** [comment]
but actually no, this "Datengrab_Bilder" only holds 409 pictures

**Comment by AxelG-DE** [comment]
I know, I am fingerpointing to my self but it is what it is....

My laptop is a comparably slow lenovo T450s with many things exactly the same than PC 
- gentoo, up to date
- dt-master up to date
- same gtk+ version
- same kernel version 

what is different:
- only one ssd
- much smaler directory
- no nvidia
- no lvm-cache

**Finding:**
no matter what I tried, even point the import-dialog to "/", I **cannot reproduce** there

**Comment by AxelG-DE** [comment]
just in case, someone wonders, why I do lvm-cache
(me shows off :-D )
![grafik](https://github.com/darktable-org/darktable/assets/49004103/e4348f6f-ba18-452b-839e-3ad2b4a45e67)

**Comment by zisoft** [comment]
I am still convinced there must be something wrong with your directory structure.

What I would do now:

Move all folders out of the parent folder "Datengrab_Bilder" and import the folder. If that works, start moving back the folders one by one and try to import until you find the folder which causes the problem.

**Comment by AxelG-DE** [comment]
> Move all folders out of the parent folder "Datengrab_Bilder" and import the folder. If that works, start moving back the folders one by one and try to import until you find the folder which causes the problem.

How could I import an empty dir?
I skip that as I do have the same issue also on the main picture folder. What I may try tomorrow is transfer that Datengrab_Bilder to my laptop (just tomorrow other tasks have priority)

> A file system check is also worth a try.
Done!
No findings.

You know, during my bisect, there was no any picture imported yet and I can clearly discriminate between good and bad. But yeah I wish I could find the things more tangible....

**Comment by TurboGit** [comment]
For the record I have double checked that the hidden directories are properly skipped.

Yes, if you have recursive ticked and you select your $HOME directory darktable will parse your whole $HOME and  this can take quite some time. More than a minute for me and during this time the UI is frozen. The only solution to that is to have a background process for the parsing and dynamically updating the dialog with a way to stop the parsing. I'll open a ticket.

**Comment by AxelG-DE** [comment]
keep in mind, on my laptop I even chose "/" and the behaviour was not quite the same. 

Anyhow I get from your message, somehow you can also create that modern-art when moving around the dialog in a frozen GUI

How big in GB or TB is your home-dir (provided it holds the pictures)?
Actually I do not purposely point to my home-dir :smile:

**Comment by TurboGit** [comment]
> How big in GB or TB is your home-dir (provided it holds the pictures)?

It is 1TB of SSD. I have very few pictures in it but quite some files/directories from devel on different projects.

**Comment by TurboGit** [comment]
@AxelG-DE : You may want to test #15894.

**Comment by AxelG-DE** [comment]
> @AxelG-DE : You may want to test #15894.

Tested on 18th Dec in the morning. Thousand times better now. I feel we still have a way to go.

**Comment by AxelG-DE** [comment]
Finally we pushed #15894 today to something super smooth. Once that is merged, this issue can be closed IMHO (or we hold it open until 4.6.1. that is the decision of the "release director" :smile:

**Comment by AxelG-DE** [comment]
@TurboGit we are working on other things near to this one, but that particular topic is solved. I close it now. You may re-open if you think differently :)
